apps/src/templates/InlineDropdownMenu.jsx

```diff
@@ -0,0 +1,143 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import Radium from 'radium';
+import color from '@cdo/apps/util/color';
+import onClickOutside from 'react-onclickoutside';
+
+// HTML Adapted from:
+// https://www.w3.org/TR/wai-aria-practices-1.1/examples/menu-button/menu-button-actions.html
+
+// Displays an accessible dropdown menu with actions or links
+// TODO: Convert to a functional component once Radium is upgraded past 0.26.0
+// Our version of Radium doesn't support useState, but it has been patched in
+// later versions
+export const InlineDropdownMenu = class InlineDropdownMenuComponent extends Component {
+  static propTypes = {
+    icon: PropTypes.string.isRequired,
+    children: props => {
+      React.Children.map(props.children, child => {
+        if (child.type !== 'a') {
+          throw new Error('only accepts children of type <a/>');
+        }
+        if (!child.props.href && !child.props.onClick) {
+          throw new Error('each child must have an href or onclick');
+        }
+      });
+    }
+  };
+
+  state = {
+    isOpen: false
+  };
+
+  handleClickOutside = () => {
+    if (this.state.isOpen) {
+      this.setState({isOpen: false});
+    }
+  };
+
+  selectOptionWrapper = selectAction => {
+    this.setState({isOpen: false});
+    selectAction();
+  };
+
+  handleKeyDown = (event, selectAction) => {
+    switch (event.which) {
+      // 13 is "enter" and 32 is "space"
+      case 13:
+      case 32: {
+        this.selectOptionWrapper(selectAction);
+      }
+    }
+  };
+
+  render() {
+    const {children, icon} = this.props;
+    const {isOpen} = this.state;
+    if (children === undefined || children.length === 0) {
+      return;
+    }
+
+    return (
+      <>
+        <button
+          type="button"
+          style={styles.menuButton}
+          onClick={() =>
+            this.setState({
+              isOpen: !isOpen
+            })
```

Comment:
it's somewhat unlikely to happen, but this is actually a race condition because we're saving a reference to `isOpen`, which may be different when react actually executes this state change. you can prevent this by using the parameters of the updater function rather than passing an object to `setState`: https://reactjs.org/docs/react-component.html#setstate

specifically, this example in that part of the docs:
```javascript
this.setState((state, props) => {
  return {counter: state.counter + props.step};
});
```

Reply:
Thank you!!